Formatted the string in missing space dialog

souravmunjal opened the pull request:
…formatter error dialog

## Pull Request template

## Purpose / Description
Had to fix the format of unreadable error message about the missing space message dialog

## Fixes
Fixes #9954  

## Approach
simply formatted the issue and generated a new string corresponding to that error i.e import_log_insufficient_space

## How Has This Been Tested?
This can be simply tested by even changing a small if condition (i.e. uncompressedSize > availableSpace)  in AnkiPackageImporter.java and importing a normal size .apkg zip file from the import option of the menu of the home page of the Ankidroid app.
or 
try to simple import a large .apkg package whose size is greater than the free space of the device


## Checklist
_Please, go through these checks before submitting the PR._

- [ done] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [ done] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ done] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [ done] You have commented your code, particularly in hard-to-understand areas
- [ done] You have performed a self-review of your own code
- [ done] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- 
![image](https://user-images.githubusercontent.com/30770269/144576175-d26c1227-1231-4490-88ac-75573df60713.png)
here i have updated the string in just base language and for the other language i have not configured this string as i thought that simply translating this string to other languages would change it's context w.r.t. to that language so configuring this string in different language is pending.

mikehardy reviewed (changes requested):
Great, thank you for posting this!
Can you remove all but the english translation per https://github.com/ankidroid/Anki-Android/wiki/Development-Guide#handling-translations ?
The other translations are handled on crowdin, our translation platform, and will simply be deleted the next time we resync, but it makes the diff enormous so it is hard to review

mikehardy commented:
If you update your description here on the PR and change the 'Fixes' area to say `Fixes #issuenumber` github automatically links the issue and PR together and will automatically close the issue when the PR is merged

souravmunjal commented:
> Great, thank you for posting this! Can you remove all but the english translation per https://github.com/ankidroid/Anki-Android/wiki/Development-Guide#handling-translations ? The other translations are handled on crowdin, our translation platform, and will simply be deleted the next time we resync, but it makes the diff enormous so it is hard to review

hey mike , i have already removed all the instances of the string in other languages and just kept the english instance in values/string base file

mikehardy commented:
@souravmunjal that does not match the visible state of the pull request: https://github.com/ankidroid/Anki-Android/pull/9965/files

souravmunjal commented:
> @souravmunjal that does not match the visible state of the pull request: https://github.com/ankidroid/Anki-Android/pull/9965/files

so what i did was that the original string was "import_log_insufficient_space" and had translations in all languages now what I did was i deleted all the instances of that string in all the language files and just kept the one instance in the base file i.e. " AnkiDroid/src/main/res/values/02-strings.xml " i.e. the last file in the PR  i.e. what you want me to do right?

mikehardy commented:
From https://github.com/ankidroid/Anki-Android/wiki/Development-Guide#handling-translations

> If you update the value of an existing key it will not be marked as needing translation. Existing translations are preserved. If you need to change a string significantly, meaning it needs re-translation, you must use a new key and remove the old key.

This seems to match that case. Needs a delete of the old key and a new key name so it will receive new translations

Sorry I'm being so brief - I'm behind deadline on a couple projects so just repyling literally as fast as possible :weary:

souravmunjal commented:
> From https://github.com/ankidroid/Anki-Android/wiki/Development-Guide#handling-translations
> 
> > If you update the value of an existing key it will not be marked as needing translation. Existing translations are preserved. If you need to change a string significantly, meaning it needs re-translation, you must use a new key and remove the old key.
> 
> This seems to match that case. Needs a delete of the old key and a new key name so it will receive new translations
> 
> Sorry I'm being so brief - I'm behind deadline on a couple projects so just repyling literally as fast as possible 😩

no problem man got your point will make the changes and generate a new pr

david-allison-1 commented:
The pull request still shows 94 files changed, did you push your changes?

souravmunjal commented:
> The pull request still shows 94 files changed, did you push your changes?

yeah i changed the string name "import_log_insufficient_space" to "import_log_insufficient_space_error" and deleted it in all the other languages just kept it in the base string file so thats why its showing 94 changes

david-allison-1 commented:
> > The pull request still shows 94 files changed, did you push your changes?
> 
> yeah i changed the string name "import_log_insufficient_space" to "import_log_insufficient_space_error" and deleted it in all the other languages just kept it in the base string file so thats why its showing 94 changes. It is showing 2 commits btw

We only want changes to the file in `/values`. 

For the file in `/values`: Delete the old key, and add the new key. 

Don't modify the translated files: our scripts which sync with our translation provider handle them.

See: https://github.com/ankidroid/Anki-Android/wiki/Development-Guide#adding-translations, feel free to improve it if it's unclear for your use case.

It's not feasible for us to review the 94 files as-is.

souravmunjal commented:
updated the repo please check again

mikehardy commented on AnkiDroid/src/main/res/values/strings.xml:
```diff
@@ -6,4 +6,5 @@
     <!-- https://developers.google.com/analytics/devguides/collection/android/v4/limits-quotas -->
     <integer name="ga_sampleFrequency">10</integer>
     <bool name="ga_anonymizeIp">true</bool>
+    <string name="import_log_insufficient_space_error">There is not enough space!\nNeeded - %1$s \nAvailable - %2$s.\n\nHaving Trouble! Sync uses less space than import. You may try importing the package on some other device (for eg. Anki Desktop) and then sync it to this device.</string>
```
```suggestion
    <string name="import_log_insufficient_space_error">There is not enough space to import the package.\nNeeded - %1$s \nAvailable - %2$s.\n\nHaving Trouble? Try to free space on your phone if possible. Alternatively, sync uses less space than import. You may try importing the package a device with more space (for example Anki Desktop) and then sync it to this device.</string>
```

just some wording changes, what do you think?

souravmunjal replied:
sure we can use updated the repo with the above string please check

david-allison-1 commented on AnkiDroid/src/main/res/values/02-strings.xml:
```diff
@@ -154,7 +154,7 @@
     <string name="import_log_failed_copy_to">Failed to copy apkg to %s</string>
     <string name="import_log_failed_validate">apkg failed validation</string>
-    <string name="import_log_insufficient_space">There is not enough space to unzip the package. Needed %1$d, available %2$d</string>
+    <string name="import_log_insufficient_space_error">There is not enough space to import the package.\nNeeded - %1$s \nAvailable - %2$s.\n\nHaving Trouble? Try to free space on your phone if possible. Alternatively, sync uses less space than import. You may try importing the package a device with more space (for example Anki Desktop) and then sync it to this device.</string>
```
> You may try importing the package a device

Needs a "using/with/via/onto" between "package" and "a".

Proposed choice: "with"

souravmunjal replied:
done please check again

mikehardy reviewed (approved):
I like it!

david-allison-1 commented:
Thanks!!

![](https://media.giphy.com/media/BYoRqTmcgzHcL9TCy1/giphy.gif)

Arthur-Milchior commented:
I fear that we may run into a problem if the user does not think about deleting the .apkg they downloaded. But I don't see how to add this information or any other more detailed explanation without having a very long and so complex error message. Even a link to a webpage would be complex (not counting the fact that it is usually not translated)

Anyway, thanks a lot for closing my issue :)

mikehardy commented:
Hi there! December OpenCollective Notice

Just a friendly notice that we try to process OpenCollective payments monthly - it's time for December 2021 submissions

If you are interested in compensation for this work, the process with details is here:

https://github.com/ankidroid/Anki-Android/wiki/OpenCollective-Payment-Process#how-to-get-paid

(I only post one comment per person to avoid spamming you, regardless of the number of PRs merged, but this note applies to all PRs merged for in the month of December)

Thanks!
